core/modules/catalog/store/stock/actions.ts

```diff
@@ -1,196 +1,73 @@
+      const { result, task_id } = await StockService.check(product.sku)
       return {
-        qty: task.result ? task.result.qty : 0,
-        status: task.result
-          ? task.result.is_in_stock
-            ? 'ok'
-            : 'out_of_stock'
-          : 'ok',
-        onlineCheckTaskId: task.task_id
-      } // if online we can return ok because it will be verified anyway
-    } else {
-      return {
-        qty: product.stock ? product.stock.qty : 0,
-        status: product.stock
-          ? product.stock.is_in_stock
-            ? 'ok'
-            : 'out_of_stock'
-          : 'volatile'
-      } // if not online, cannot check the source of true here
+        qty: result ? result.qty : 0,
+        status: getStatus(result, 'ok'),
+        onlineCheckTaskId: task_id
+      }
+    }
+
+    return {
+      qty: product.stock ? product.stock.qty : 0,
+      status: getStatus(product, 'volatile')
     }
   },
-  /**
-   * Reset current configuration and selected variatnts
-   */
-  list ({ commit }, { skus }) {
-    if (config.stock.synchronize) {
-      try {
-        const task: any = TaskQueue.execute({
-          url: processURLAddress(
-            `${config.stock.endpoint}/list?skus=${encodeURIComponent(
-              skus.join(',')
-            )}`
-          ),
-          payload: {
-            method: 'GET',
-            headers: { 'Content-Type': 'application/json' },
-            mode: 'cors'
-          },
-          skus: skus
+  async list ({ commit }, { skus }) {
+    if (!config.stock.synchronize) return
+
+    const task = await StockService.list(skus)
+
+    if (task.resultCode === 200) {
+      const productInfos = getProductInfos(task.result)
+
+      for (const productInfo of productInfos) {
+        commit(stockMutationTypes.SET_STOCK_CACHE_PRODUCT, {
+          productId: productInfo.product_id,
+          productInfo
         })
-        if (task.resultCode === 200) {
-          for (const si of task.result) {
-            const productInfo = {
-              is_in_stock: si.is_in_stock,
-              qty: si.qty,
-              product_id: si.product_id
-            }
-            commit(stockMutationTypes.SET_STOCK_CACHE_PRODUCT, {
-              productId: si.product_id,
-              productInfo
-            })
-          }
-        }
-        return task // if online we can return ok because it will be verified anyway
-      } catch (err) {
-        Logger.error(err, 'stock')()
-        return null
       }
-    } else {
-      return null // if not online, cannot check the source of true here
     }
+
+    return task
   },
   clearCache ({ commit }) {
     commit(stockMutationTypes.SET_STOCK_CACHE, {})
-  },
-  async stockAfterCheck ({ dispatch, commit }, event) {
-    setTimeout(async () => {
-      // TODO: Move to cart module
-      const cartItem: any = await dispatch('cart/getItem', event.product_sku, {
-        root: true
-      })
-      if (cartItem && event.result.code !== 'ENOTFOUND') {
```

Comment:
This check - ENOTFOUND was for a offline mode case; I couldn’t find it anywhere in the new code?

Reply:
Basically I wanted to ask about that, because I used instead `result.code !== 404`:  (https://github.com/DivanteLtd/vue-storefront/pull/3395/files#diff-729df756e1596492966011c1249d3b8bR80) - I'll change it